projects/ngx-sub-form/src/lib/ngx-sub-form.component.ts

```diff
@@ -4,42 +4,16 @@ import { Subscription } from 'rxjs';
 import { delay, tap } from 'rxjs/operators';
 import { Controls, ControlsNames, getControlsNames } from './ngx-sub-form-utils';
 
-export class FormControlsRequiredError extends Error {
-  message = `Passing the "formControls" is required to have NgxSubForm working`;
-}
-
 export abstract class NgxSubFormComponent<ControlInterface, FormInterface = ControlInterface>
   implements ControlValueAccessor, Validator, OnDestroy {
-  protected abstract formControls: Controls<FormInterface>;
-
   public get formControlNames(): ControlsNames<FormInterface> {
-    return getControlsNames(this.formControls);
+    return getControlsNames(this.getFormControls());
   }
 
-  // flag to know whether the component has been destroyed or not because there's
-  // currently a memory leak when using custom ControlValueAccessor
-  private destroyed = false;
-
-  private fg: FormGroup | undefined;
+  public formGroup: FormGroup;
 
-  public get formGroup(): FormGroup {
-    if (this.destroyed) {
-      // we do not want to type the formGroup as `FormGroup | undefined`
-      // because it will require to handle that use case from every component (ts and html)
-      // just to get the typings right, whereas we know formGroup will be defined when needed
-      // and we're just trying to avoid the memory leak issue. This shouldn't affect the public API
-      return undefined as any;
-    }
-
-    if (!this.formControls) {
-      throw new FormControlsRequiredError();
-    }
-
-    if (!this.fg) {
-      this.fg = new FormGroup(this.formControls);
-    }
-
-    return this.fg;
+  constructor() {
+    this.formGroup = new FormGroup(this.getFormControls());
```

Comment:
This can just be direct in the declaration

Reply:
Done